pkg/bootstrap/platform/k8s.go

```diff
@@ -309,8 +318,60 @@ func K8SConfig(config *Config) ([]byte, error) {
 	var k8sConfig bytes.Buffer
 	t := template.Must(template.New("").Parse(k8sTemplate))
 	if err := t.Execute(&k8sConfig, args); err != nil {
-		return nil, fmt.Errorf("cannot create k8s configuration off template: %w", err)
+		return nil, fmt.Errorf("cannot create k8s configuration from template: %w", err)
+	}
+
+	certConfig, err := K8SCertificateConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	k8sBytes := certConfig
+	k8sBytes = append(k8sBytes, k8sConfig.Bytes()...)
+	return k8sBytes, nil
+}
+
+// K8SCertificateConfig returns a kubernetes secrets that contains all the certificates.
+func K8SCertificateConfig(config *Config) ([]byte, error) {
+	args := map[string]interface{}{
+		"fabricCA":         base64.StdEncoding.EncodeToString(config.FabricCertificate.RawCert()),
+		"peerCA":           base64.StdEncoding.EncodeToString(config.PeerCertificate.RawCert()),
+		"controlplaneCert": base64.StdEncoding.EncodeToString(config.ControlplaneCertificate.RawCert()),
+		"controlplaneKey":  base64.StdEncoding.EncodeToString(config.ControlplaneCertificate.RawKey()),
+		"dataplaneCert":    base64.StdEncoding.EncodeToString(config.DataplaneCertificate.RawCert()),
+		"dataplaneKey":     base64.StdEncoding.EncodeToString(config.DataplaneCertificate.RawKey()),
+		"gwctlCert":        base64.StdEncoding.EncodeToString(config.GWCTLCertificate.RawCert()),
+		"gwctlKey":         base64.StdEncoding.EncodeToString(config.GWCTLCertificate.RawKey()),
+	}
+
+	var certConfig bytes.Buffer
+	t := template.Must(template.New("").Parse(certsTemplate))
```

Comment:
IIUC, all secrets are added as inlined byte arrays (base64 of the cert). Have you considered using k8s object references instead, so that the actual credentials are in secrets?

Reply:
As we spoke secrets are not part of the operator, but part of the controlplane/dataplane we can open an issue about that. (@orozery)

Reply:
yes - please do open an issue. If the secret are added as independent object (which I think they should be since that would allow for easier rotation by existing tools), then we should track this incomplete item